edb/server/protocol/auth_ext/data.py

```diff
@@ -80,3 +80,126 @@ def __repr__(self) -> str:
             f"iss={self.iss!r} "
             f"email={self.email!r})"
         )
+
+
+@dataclasses.dataclass
+class OpenIDConfig:
+    """
+    OpenID Connect configuration.
+    See:
+    - https://openid.net/specs/openid-connect-discovery-1_0.html
+    - https://accounts.google.com/.well-known/openid-configuration
+    """
+
+    issuer: str
+    authorization_endpoint: str
+    token_endpoint: str
+    userinfo_endpoint: Optional[str]
+    jwks_uri: str
+    registration_endpoint: Optional[str]
+    scopes_supported: Optional[list[str]]
+    response_types_supported: list[str]
+    response_modes_supported: Optional[list[str]]
+    grant_types_supported: Optional[list[str]]
+    acr_values_supported: Optional[list[str]]
+    subject_types_supported: list[str]
+    id_token_signing_alg_values_supported: list[str]
+    id_token_encryption_alg_values_supported: Optional[list[str]]
+    id_token_encryption_enc_values_supported: Optional[list[str]]
+    userinfo_signing_alg_values_supported: Optional[list[str]]
+    userinfo_encryption_alg_values_supported: Optional[list[str]]
+    userinfo_encryption_enc_values_supported: Optional[list[str]]
+    request_object_signing_alg_values_supported: Optional[list[str]]
+    request_object_encryption_alg_values_supported: Optional[list[str]]
+    request_object_encryption_enc_values_supported: Optional[list[str]]
+    token_endpoint_auth_methods_supported: Optional[list[str]]
+    token_endpoint_auth_signing_alg_values_supported: Optional[list[str]]
+    display_values_supported: Optional[list[str]]
+    claim_types_supported: Optional[list[str]]
+    claims_supported: Optional[list[str]]
+    service_documentation: Optional[str]
+    claims_locales_supported: Optional[list[str]]
+    ui_locales_supported: Optional[list[str]]
+    claims_parameter_supported: Optional[bool]
+    request_parameter_supported: Optional[bool]
+    request_uri_parameter_supported: Optional[bool]
+    require_request_uri_registration: Optional[bool]
+    op_policy_uri: Optional[str]
+    op_tos_uri: Optional[str]
+
+    def __init__(self, **kwargs):
+        for field in dataclasses.fields(self):
+            setattr(self, field.name, kwargs.get(field.name))
```

Comment:
We could probably store the unused remaining values from `kwargs` into a single dict field for debugging purposes.

Reply:
@fantix ⚡ [`d6da52d` (#6013)](https://github.com/edgedb/edgedb/pull/6013/commits/d6da52d1206ad2a8b655b0a90267754cf1e35625)

Reply:
Cool! It's actually okay to just `self._extra_fields = kwargs` but this is trivial, pls ignore it.

Reply:
Oh, no I think this is more obvious what it's doing, will fix!